### PR TITLE
Add `pants-plugins/pack_metadata` to inform pants about our packs

### PR DESCRIPTION
**Pull request by cognifloyd**
### Background

This is another part of introducing [pants](https://www.pantsbuild.org/docs), as discussed in various TSC meetings.

Related PRs can be found in:
- [pantsbuild](https://github.com/StackStorm/st2/milestone/47) milestone (which includes PRs since #5713)
- https://github.com/StackStorm/st2/labels/pantsbuild label (which also includes preparatory PRs that came before adding pantsbuild)

### Overview of this PR

This PR makes it possible to inform pants about all of our pack metadata files by adding a new `pack_metadata` target. The `pack_metadata` target encompasses the following files:

https://github.com/StackStorm/st2/blob/185e871260000b579902c9de449204f905959f2f/pants-plugins/pack_metadata/target_types.py#L29-L40

Python, shell, and requirements files all need to be handled separately so that pants can lint/fmt them appropriately. Pants does not have a markdown plugin yet, so I put them in the list as a reminder that they will need to be dealt with, but I commented them out.

One of our packs is in a git submodule (`st2tests/st2tests/fixtures/packs/test_content_version`), so it also adds a `pack_metadata_in_git_submodule` target to cleanly handle a case where someone has not checked out the git submodules. This improves the Developer Experience by raising an error as early as possible to let people know that they need to checkout the git submodule and how to do that. Otherwise, they won't get an error until they run tests where we have some code to give the same instructions for the tests that need it.

Note that I have not enabled the plugin yet:
https://github.com/StackStorm/st2/blob/185e871260000b579902c9de449204f905959f2f/pants.toml#L28

That keeps this PR focused on the plugin itself including adding tests for it. The next PR will add the `pack_metadata` targets (at least in part using `./pants tailor ::`) in our BUILD files. If you want to explore some of that, you can look at: https://github.com/StackStorm/st2/compare/pants-plugins-pack_metadata...pants-pack_metadata

### Developing pants plugins

Pants has extensive documentation on the plugin API, including how to create targets, how to write rules, and there's even a mini tutorial about how to add a formatter (adding formatters is a very common thing for plugins to do).

- [Plugins overview](https://www.pantsbuild.org/docs/plugins-overview)
- [The Target API](https://www.pantsbuild.org/docs/target-api)
    - [Concepts](https://www.pantsbuild.org/docs/target-api-concepts) _eg: defines targets and fields._
    - [Creating new fields](https://www.pantsbuild.org/docs/target-api-new-fields)
    - [Creating new targets](https://www.pantsbuild.org/docs/target-api-new-targets)
- [The Rules API](https://www.pantsbuild.org/docs/rules-api)
- [Testing Plugins](https://www.pantsbuild.org/docs/rules-api-testing)

### CODE REVIEW

**Review by rush-skills — APPROVED** [review]

File: pants-plugins/README.md [hunk 1]
```diff
@@ -26,6 +29,26 @@ This plugin also wires up pants so that the `lint` goal runs additional
...
 like `./pants lint st2common/st2common/openapi.yaml`.
 
+### `pack_metadata` plugin
+
+This plugin adds two new targets to pants:
+- `pack_metadata`
+- `pack_metadata_in_git_submodule`
+
+These targets include all StackStorm pack metadata files in a pack.
+Pack metadata includes top-level files (`pack.yaml`, `<pack>.yaml.examle`,
```
Comment (amanda11) [inline pants-plugins/README.md:39]:
```suggestion
Pack metadata includes top-level files (`pack.yaml`, `<pack>.yaml.example`,
```

File: pants-plugins/pack_metadata/target_types.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            )
+        super().validate_resolved_files(files)
+
+
+class PackMetadata(ResourcesGeneratorTarget):
+    alias = "pack_metadata"
+    core_fields = (*COMMON_TARGET_FIELDS, Dependencies, PackMetadataSourcesField)
+    help = (
+        "Loose pack metadata files.\n\n"
+        "Pack metadata includes top-level files (pack.yaml, <pack>.yaml.examle, "
```
Comment (amanda11) [inline pants-plugins/pack_metadata/target_types.py:62]:
```suggestion
        "Pack metadata includes top-level files (pack.yaml, <pack>.yaml.example, "
```

**Review by amanda11 — CHANGES_REQUESTED** [review]
Couple of typos

**Review by cognifloyd — COMMENTED** [review]
Good catch! Fixed.

**Review by amanda11 — APPROVED** [review]
LGTM

**Pull request merged**